goth/runner/probe/__init__.py

```diff
@@ -63,6 +69,12 @@ class Probe(abc.ABC):
     in subclasses (see `ProviderProbe` and `RequestorProbe`).
     """
 
+    agents: Set[AgentComponent]
+    """Set of agent components that will be started as part of this probe."""
```

Comment:
I agree, this could be useful. Please see my comment on `AgentComponent.__hash__`. We could replace this with an `OrderedSet` (though that would require pulling in an additional dependency, i.e. https://pypi.org/project/ordered-set/) or an `OrderedDict`.

Reply:
After your suggestion on `AgentComponent.__hash__` I'm going to change this to an `OrderedDict`.